visionmetrics/caption/azure_openai_model_eval_base.py

```diff
@@ -0,0 +1,161 @@
+import torch
+from collections import Counter
+from enum import Enum
+from irisml.tasks.create_azure_openai_chat_model import OpenAITextChatModel
+from torchmetrics import Metric
+
+
+# Prompt template placeholder constants
+PREDICTION_PLACEHOLDER = "<|prediction|>"
+TARGET_PLACEHOLDER = "<|target|>"
+
+# Response parsing constants
+MULTIPLE_RESPONSES_DELIMITER = "<|delimiter|>"
+
+# Evaluator separator constants for joining multiple ground truth values
+OR_SEPARATOR = " <|OR|> "
+
+
+# Enum for per-value result status type
+class ResultStatusType(Enum):
+    TruePositive = 1
+    TrueNegative = 2
+    FalseNegative = 3
+    FalsePositiveGtNotNull = 4
+    FalsePositiveGtNull = 5  
+
+
+class AzureOpenAITextModelCategoricalEvalBase(Metric):
+    """
+    Evaluator that uses single-turn text-only calls to an Azure OpenAI model for evaluation and returns
+    standard categorical metrics (precision, recall, F1, accuracy) based on the scores returned.
+
+    Args:
+        prompt_template: string template for the prompt instructing how to score the prediction. Must have two placeholders: <|prediction|> and <|target|>.
+        positive_threshold: determines the threshold (inclusive) at which the score assigned by the model to the prediction means the prediction is correct. Range: [0.0, 1.0].
+        negative_value: indicates the value (e.g., None, 'null', 0, '', []) for which predicted and target values are considered null. Note that the evaluator strictly requires
+                        an exact match for either a prediction or target to be considered negative. TODO: Implement fuzzy negative matching. Scores are always computed so they
+                        can be used for average_score computation.
+
+        Other parameters follow the standards of irisml.tasks.create_azure_openai_chat_model.OpenAITextChatModel.
+    """
+    def __init__(self, endpoint: str, deployment_name: str, system_message: str, prompt_template: str,
+                 temperature=0.0, max_tokens=50, requests_interval=30, num_responses=1, positive_threshold=0.5, negative_value=''):
+        super().__init__()
+        if PREDICTION_PLACEHOLDER not in prompt_template or TARGET_PLACEHOLDER not in prompt_template:
+            raise ValueError("Both the predicted placeholder {PREDICTION_PLACEHOLDER} and target placeholder {TARGET_PLACEHOLDER} must be present in prompt_template.")
```

Comment:
If both placeholders must be present, then maybe you meant to use AND in line #45?